lib/skate_web/controllers/detours_controller.ex

```diff
@@ -4,10 +4,25 @@ defmodule SkateWeb.DetoursController do
   alias Skate.OpenRouteServiceAPI
   use SkateWeb, :controller
 
+  alias Skate.Detours.Detours
   alias Skate.Detours.MissedStops
   alias Skate.Detours.RouteSegments
+  alias SkateWeb.AuthManager
   alias Util.Location
 
+  @spec update_snapshot(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def update_snapshot(conn, %{"snapshot" => snapshot} = _params) do
+    uuid = Map.get(snapshot, "uuid")
+
+    %{id: user_id} = AuthManager.Plug.current_resource(conn)
+
+    Detours.update_or_create_detour_for_user(user_id, uuid, %{
+      state: snapshot
+    })
+
+    send_resp(conn, 200, "")
```

Comment:
thought: along the lines of my [comment](https://github.com/mbta/skate/pull/2729/files#r1704411158), I would suggest that if `uuid` is not in the `context`, then we insert and return the UUID back to the frontend for the state machine to save. What do you think

Reply:
Did a variation of this: always return the UUID, but only sometimes save it in the state machine. In hindsight, unsure if I took the right approach?